polygon_client/client.go

```diff
@@ -75,112 +89,71 @@ func (c *Client) LoadTickerData(ctx context.Context, tickerSymbol string) (*mode
 	return ticker, nil
 }
 
-func (c *Client) GetTickerTodayAggs(ctx context.Context, t time.Time, ticker string, rangeSize int) (results []*models.Agg, err error) {
+func (c *Client) GetTickerTodayAggs(ctx context.Context, t time.Time, ticker string, rangeSize int) ([]*models.Agg, error) {
 	loc, _ := time.LoadLocation("America/New_York")
 
 	// Start at 9am instead of 930am because sometimes pre market is significant to the charts.
 	openTime := time.Date(t.Year(), t.Month(), t.Day(), 9, 0, 0, 0, loc)
 	closeTime := time.Date(t.Year(), t.Month(), t.Day(), 16, 30, 0, 0, loc)
 
-	url := fmt.Sprintf("https://api.polygon.io/v2/aggs/ticker/%s/range/%d/minute/%d/%d?apiKey=%s",
-		ticker,
-		rangeSize,
-		(openTime.UnixNano() / int64(time.Millisecond)),
-		(closeTime.UnixNano() / int64(time.Millisecond)),
-		c.APIKey,
-	)
-	body, err := makeHTTPRequest(ctx, url)
-	if err != nil {
-		return results, err
-	}
+	aggsParams := polygon_models.GetAggsParams{
+		Ticker:     ticker,
+		Multiplier: rangeSize,
+		Resolution: polygon_models.Minute,
+		From:       polygon_models.Millis(openTime),
+		To:         polygon_models.Millis(closeTime),
+	}.WithLimit(int(closeTime.Sub(openTime) / time.Minute))
 
-	res := &AggsResponse{}
-	if err := json.Unmarshal(body, res); err != nil {
-		return results, fmt.Errorf("unable to parse JSON response from polygon: %w", err)
+	resp, err := c.restClient.GetAggs(ctx, aggsParams)
+	if err != nil {
+		return nil, err
 	}
 
 	// Transform our polygon.io aggregates into the "model" aggregates.
-	for _, agg := range res.Results {
-		newAgg := &models.Agg{
+	results := make([]*models.Agg, 0, len(resp.Results))
```

Comment:
any particular reason why you do this copy operation?

Reply:
I was trying to avoid touching other parts of the code for now. But I don't know if I can avoid this entirely, since the `models.Agg` is a proto model that the server sends to its clients

Reply:
oh gotcha, that makes sense then